mlir/lib/Gradient/Transforms/ConversionPatterns.cpp

```diff
@@ -223,10 +226,15 @@ struct BackpropOpPattern : public ConvertOpToLLVMPattern<BackpropOp> {
                     callArgs.push_back(castToConvertedType(arg, rewriter, loc));
                 }
             }
-            else {
+            else if (isa<MemRefType>(arg.getType())) {
                 size_t position = std::distance(diffArgIndices.begin(), it);
                 unpackMemRef(arg, dataIn[position], callArgs, rewriter, loc, {.zeroOut = true});
             }
+            else {
+                assert(isa<FloatType>(arg.getType()) && "diff arg must be a float or float tensor");
+                callArgs.push_back(arg);
+            }
+            index++;
```

Comment:
Is this increment intentional (skipping one iteration) given that `index` enumerates the args?

Reply:
This is definitely a bug that was from a merge conflict artifact. I'll remove it.